File: 2nd Year/1st Semester/MAP/Laborator/Lab9/Lab9/model/BubbleSort.cs
namespace Lab9.model;

public class BubbleSort : AbstractSorter
{

    public override void Sort(int[] array)
    {
        var nr = array.Length;
        for (var i = 0; i < nr; i++) {
            for (var j = 0; j < nr - 1; j++) {
                if (array[j] > array[j + 1]) {
                    (array[j], array[j+1]) = (array[j+1], array[j]);
                }
            }
        }
    }
}

File: 2nd Year/1st Semester/MAP/Laborator/SEM1_LAB2_MAP/model/BubbleSort.java
package sem1_2.model;

import sem1_2.model.AbstractSorter;

public class BubbleSort extends AbstractSorter {
    @Override
    public void sort(int[] array) {
        int nr = array.length;
        for (int i = 0; i < nr; i++) {
            for (int j = 0; j < nr - 1; j++) {
                if (array[j] > array[j + 1]) {
                    int aux = array[j];
                    array[j] = array[j+1];
                    array[j+1] = aux;
                }
            }
        }
    }
}